packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -245,8 +264,14 @@ export class WindowsOS extends OperatingSystem {
  * OS features specialized for Linux
  */
 export class LinuxOS extends OperatingSystem {
-  public createUserData(scripts: string[]): string {
-    return '#!/bin/bash\n' + scripts.join('\n');
+  public createUserData(scripts: string[], options: UserDataOptions): string {
+    const bashFlags = [];
+    if (options.verbose !== false) { bashFlags.push('x'); }
+    if (options.strict) { bashFlags.push('eu'); }
```

Comment:
By the way, I'd expect `strict` to also imply `pipefail`.

Reply:
Sure. I was on the fence for that one, but I can totally add it.